### DISCUSSION TITLE
sBTC Deposit - Diagram and Flow

### DISCUSSION THREAD

**Discussion by 8marz8** [stacks-sbtc/sbtc#8] [Research]
Separate discussion topic to consider possible Deposit/Peg-In/Mint flows and settle on a near-final diagram that captures our HLD decision for sBTC V1. 

## Deposit [Diagram](https://www.figma.com/file/owHg2NP4kmqNmUc3jynkgB/sBTC-Deposit-(Copy)?type=whiteboard&node-id=0-1&t=Bdh7x7Nx6pXTT6pZ-0) For API-Only Ingress

![sBTC Deposit (Copy) (3)](https://github.com/Trust-Machines/sbtc-v1/assets/132940173/47ef0ef1-d352-4cbd-8acb-aab4a16a7c0e)

## Deposit Flow For API-Only Ingress
1. **Deposit request**: A bitcoin holder creates a transaction on Bitcoin.
    * The deposit transaction contains a UTXO (**deposit UTXO**) spendable by sBTC Signers, with an OP_DROP payload.
    * The payload contains the recipient address of the sBTC among other relevant info for the deposit.
       * The relevant info could contain fee suggestion or max_fee
2. **Proof of deposit**: The bitcoin holder submits a proof of deposit on Stacks by invoking the Signer API
3. **Deposit accept**: 
    * **Deposit redeem**: The sBTC Signers redeem the deposit by consuming the deposit UTXO, consolidating it into the sBTC UTXO.
    * **Mint**: The sBTC Signers finalize the deposit acceptance by calling the `mint` function in `.sbtc` to mint the sBTC.

This flow only requires one contract call to `mint` sBTC:
(Note: the parameters and description is not finalized/accurate)
| Function name | Parameters | Description |
| --- | --- | --- |
| `mint` | `(amount, destination, deposit-txid, burn-chain-height, merkle-proof, tx-index, block-header)` | This function takes in some information that is all read directly from the provided Bitcoin transaction. It includes the Stacks principal to mint the sBTC to, and all of the required Bitcoin transaction information required to verify it. |

---------------
Original draft diagram Link -> https://www.figma.com/file/wWLNDHKYQJjcNAlIyDHDwn/sBTC-Peg-In?type=whiteboard&node-id=0%3A1&t=kHawzKr0zVbPkskV-1

**Comment by netrome** [comment]
Nit: Can we call it "deposit" flow instead of "peg-in" to be consistent with naming?

**Comment by AshtonStephens** [comment]
1. Consensus level challenges to enforce that signers include deposits can only be included in sBTC v2 because there will be no changes to the stacks-core for the core Nakamoto signers.
2. The sBTC Signers will be the sole gatekeeper to accepting or rejecting sBTC transactions, and the challenge will still be present but not enforced

**Comment by AshtonStephens** [comment]
Question:
- Does the clarity contract need to reveal the stacking transactions

**Comment by 8marz8** [comment]
Question:
Can we confirm the API layer name? (Revealer or Signer API)?
Also, will this be part of the Signer binary, or will it be a separate one? (I can adjust the diagram accordingly and potentially conclude the other discussion topic on Revealer API)

**Reply by netrome** [reply]
I like to call it Signer API.

I'm open to other suggestions, but would prefer to refrain from the term "Revealer API", because it comes from the [previous commit-reveal documentation](https://stacks-network.github.io/sbtc-docs/sbtc-operations/commit-reveal-system.html) where the Revealer was seen as a separate role from Signers.

**Comment by AshtonStephens** [comment]
Proof of deposit  -> The information the signer will need to spend a deposit UTXO.

**Comment by AshtonStephens** [comment]
Mint call will have the same signature requirements expanded upon in https://github.com/Trust-Machines/sbtc-v1/issues/16
